docs/root-application.md

```diff
@@ -24,24 +30,30 @@ does not have to. Note that if a child application is declared from within anoth
 will be maintained between the child applications. Instead, the child applications will be siblings and will be mounted
 and unmounted according to their own activity functions.
 
-In order to declare a child application, call the `declareChildApplication(path, activityFunction)` api. Example:
+In order to declare a child application, call the `declareChildApplication(name, howToLoad, activityFunction)` api. Example:
 
 ```js
 import { declareChildApplication } from 'single-spa';
 
-declareChildApplication("src/app1/app1.main.js", function() {
+declareChildApplication("app1", () => System.import("src/app1/main.js"), function() {
   return window.location.hash.indexOf("#/app1/") === 0;
 });
-declareChildApplication("src/app2/app2.main.js", function() {
+declareChildApplication("app2", () => System.import("src/app2/main.js"), function() {
   return window.location.hash.indexOf("#/app2/") === 0;
 });
 ```
 
 ### Path to child application
-The first argument to `declareChildApplication` must be a string that your loader (such as SystemJS) can import.
+The first argument to `declareChildApplication` must be a string name.
+
+### Loading function
+The second argument to `declareChildApplication` must be a function that returns a promise (or an "async function").
```

Comment:
Maybe provide a link on `"async function"` to a resource about them. Maybe on of:

1. https://ponyfoo.com/articles/understanding-javascript-async-await
1. https://tc39.github.io/ecmascript-asyncawait/
1. https://jakearchibald.com/2014/es7-async-functions/